src/state.py

```diff
@@ -155,6 +156,40 @@ def _get_agent_meta_map_from_relation(
     return {unit.name: AgentMeta.from_agent_relation(relation.data[unit]) for unit in remote_units}
 
 
+class ProxyConfig(BaseModel):
+    """Configuration for accessing Jenkins through proxy.
+
+    Attributes:
+        http_proxy: The http proxy URL.
+        https_proxy: The https proxy URL.
+        no_proxy: Comma separated list of hostnames to bypass proxy.
+    """
+
+    http_proxy: typing.Optional[HttpUrl]
+    https_proxy: typing.Optional[HttpUrl]
+    no_proxy: typing.Optional[str]
+
+    @classmethod
+    def from_charm_env(cls, env: typing.Mapping[str, str]) -> typing.Optional["ProxyConfig"]:
```

Comment:
Why not directly reading the JUJU_CHARM_ proxy env vars directly?

Reply:
That does sound a bit better

Reply:
Changed to read from `os.environ` directly, thank you!